### DISCUSSION TITLE
Setting different defaults for the same language

### DISCUSSION THREAD

**Discussion by joros244** [explosion/spaCy#11872] [Help: Coding & Implementations] [answered]
Hello, I'm creating a SpaCy pipeline and customizing the stopword list as follows:

```
nlp_custom = spacy.load("es_core_news_md", disable=["tagger", "parser", "ner"])
nlp_custom.Defaults.stop_words.clear()
nlp_custom.Defaults.stop_words |= {'el', 'la'}
```

This works as expected. However, it seems like I'm changing globally the class Defaults so that, if I'd like to create a new pipeline with the default stopword set:
```
nlp_number_2 = spacy.load("es_core_news_md", disable=["tagger", "parser", "ner"])
```
it would not be possible. It would be affected by the stopword list modification above.

Is there a way to make this change affect only the instance and not the class? How can I have two instances with different parameters?

Thanks in advice.

**Comment by adrianeboyd** [comment, marked as answer]
Changing the defaults does affect the whole class. Instead, you'd want to use a custom language, as shown here: https://spacy.io/usage/linguistic-features/#language-subclass

If you're not using any of the components in `es_core_news_md`, you can consider using `spacy.blank("es")` instead, which would turn into `spacy.blank("custom_es")` for your customized language.

If you do want to combine components from `es_core_news_md` and your custom language class it's a little more complicated, but let me know what you're trying to do and I can explain the details.

**Reply by joros244** [reply]
Thanks for the answer, but I think that is not what I need. This is what I have done so far: 
I have created a class that provides methods to perform some common NLP tasks. Internally, this class uses a SpaCy pipeline for steps like lemmatization or stopword removal. The user can customize the behavior by, for example, providing a custom stopword list. This is done in the constructor with the approach in the message above.

Up to this point, everything works fine. However, if the user wants to instantiate two different objects with different stopword lists this would not work with our current approach, since any changes to the Deafult properties would be shared by both instances.

**Reply by adrianeboyd** [reply]
How are you using the stop word filtering? If it's not necessary to use `IS_STOP` for speed reasons (in the Matcher or as a model tok2vec feature), then it might be easier to do the stop word filtering with a simple word list outside of `token.is_stop` instead of using the language defaults?

**Reply by joros244** [reply]
Thank you, that worked fine.
